### ISSUE TITLE
w3id.org dns seems to have < 1 year to go

### ISSUE THREAD

**Issue by danbri** [perma-id/w3id.org#81]
If this is meant to last perpetually, shouldn't its dns be prepaid for decades rather than months?

**Comment by msporny** [comment]
Yes, we intend to do that. It hasn't happened yet for two reasons:

1. We didn't want to pre-buy for decades until it was clear that the website would be useful to other people. It seems to be gaining traction, but I'm not sure we're there quite yet.
1. We're waiting on funds from a non-profit to pre-buy the domain and SSL certificates. That is, unless Google would like to make a donation to make that happen sooner. :P

**Comment by akuckartz** [comment]
I am more concerned about the live expectancy of the organization behind webid.org. As long as it is alive it will be able to ensure that the domain is paid in regular intervals. And if it dies it then would not even be able to ensure that the other side of a contract does what it was bound to do.

I think it would be more important if webid.org users could automatically be notified by a dead man's switch in case webid.org dies or the domain is not paid in time.

**Comment by perma-id** [comment]
I agree that this is a concern.  But let's not over think it.  Domain
registration is very inexpensive.  I am sure it is set up to auto-renew.
Manu has some funding ideas in mind.  If people are really worried, I would
be happy to kick in $100 and pay for the next ten years.

I DO like the idea of some sort of dead man's switch.  I wonder if there is
some free domain registration monitoring service that people can use to
give some peace of mind?

On Thu, May 7, 2015 at 9:23 AM, Andreas Kuckartz <notifications@github.com>
wrote:

> I am more concerned about the live expectancy of the organization behind
> webid.org. As long as it is alive it will be able to ensure that the
> domain is paid in regular intervals. And if it dies it then would not even
> be able to ensure that the other side of a contract does what it was bound
> to do.
>
> I think it would be more important if webid.org users could automatically
> be notified by a dead man's switch in case webid.org dies or the domain
> is not paid in time.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/perma-id/w3id.org/issues/81#issuecomment-99884630>.
>



-- 
Shane McCarron
Managing Director, Applied Testing and Technology, Inc.

**Comment by prototypo** [comment]
Hi all,

Well, the entire point of w3id.org <http://w3id.org/> is that there are many organizations involved. It is true that the DNS registration is a well-known Achilles heel, in that one organization oversees it.

So, I do not worry about the life expectancy of “the organization”, but we do need to ensure that the DNS is paid. I’ll pay it if necessary. I’m sure others would, too.

Regards,
Dave
--
http://about.me/david_wood

**Comment by msporny** [comment]
Hey all,

I'm going to take care of this today. It's the last existential issue left wrt. w3id.org, so let's just fix it. Here's what I'm going to do:

1. Transfer the domain from namecheap to gandi.net.
2. Buy 10 years worth of registration for the domain.
3. Setup all the w3id.org admins as account admins for the DNS.

That should take care of this perceived problem.

To be clear, the problem was never money or institutional longevity. It was time and making sure the money was coming from the right source (a non-profit, so that we didn't need to dip into our personal bank accounts).

At this point I've spent far more time explaining to people why this isn't a problem than it would've taken to just fix the issue. So, I'm fixing the issue now. :)

**Comment by prototypo** [comment]
Thanks, Manu!

Regards,
Dave
--
http://about.me/david_wood

**Comment by danbri** [comment]
Thanks!

Maybe we  should make a buddylist of people burdened with long-lived semweb related namespace domains, and keep in touch to ensure the relevant DNS situations are remain healthy? 

Dan (xmlns.com-owner-victim since 2000:)

**Comment by kidehen** [comment]
We can make BuddyList Group right now, and right here, via responses to this comment. Simply use Nanotation to get this going. Initial salvo: 

{
<#BuddyGroup>
a foaf:Group;
foaf:member <#i> ;
is schema:about of <> . 

<#i>
a foaf:Person ;
foaf:nick "kidehen" ;
foaf:name "Kingsley Uyi Idehen" ; 
owl:sameAs <http://linkeddata.uriburner.com/about/id/entity/https/github.com/kidehen>, <http://kingsley.idehen.net/datapace/person/kidehen#this>, <https://plus.google.com/+KingsleyIdehen/about#this> ;
is schema:about of <https://twitter.com/kidehen/> , <https://plus.google.com/+KingsleyIdehen/about/>,
<https://github.com/kidehen/>, <> . 
}

**Comment by msporny** [comment]
The domain transfer is now complete... no thanks to the old provider whose domain transfer page just so happened to have a glitch in it that caused the operation to take the maximum time allowed by the rules that govern registrars. Funny how often that happens.

I'll send out the new login info to the trustees of w3id.org soon-ish.

I also tried renewing for 10 years and got some strange error:

```
2306 : Parameter value policy error (2306:Parameter value policy error (maximum_registration_period_exceeded))
```

I'll try renewing for a shorter timeframe tomorrow.

**Comment by msporny** [comment]
The w3id.org domain has been renewed for 9 years (the max allowable). The w3id.org domain expires on 2024-02-14 21:17:29

**Comment by prototypo** [comment]
Manu,

w3id.org <http://w3id.org/> is down, and it appears to be due to DNS. DNS is resolving to the registrar.

Regards,
Dave
--
http://about.me/david_wood

**Comment by scor** [comment]
For me it shows a webpage saying "This domain name has been registered with Gandi.net"

**Comment by msporny** [comment]
Yeah, on it. I told Gandhi not to muck w/ the DNS - but it did anyway. 3 hour TTL... so it'll take a while for the domain to propagate since it's already being cached elsewhere on the net. The zone file is in place... we just need to wait for it to propagate.

**Comment by msporny** [comment]
I'd just like to point out the irony that our biggest downtime for the site to date, /by far/, was due to DNS issues as a result of us trying to ensure that the site wouldn't go away due to DNS issues. :P

**Comment by danbri** [comment]
:)

gotta get worse before it gets better? thanks for wrestling with this!

**Comment by perma-id** [comment]
Hi Manu,

I don’t think the zone file is necessarily correct. I flushed my local DNS cache and received the following results from dig. Note that www.w3id.org <http://www.w3id.org/> and w3id.org <http://w3id.org/> resolve to different IPs.

Regards,
Dave
--
http://about.me/david_wood

[[
Macadamia:~ dwood$ dig w3id.org

; <<>> DiG 9.8.3-P1 <<>> w3id.org
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 33297
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;w3id.org.			IN	A

;; ANSWER SECTION:
w3id.org.		10766	IN	A	216.252.204.54

;; Query time: 2 msec
;; SERVER: 10.5.50.1#53(10.5.50.1)
;; WHEN: Wed May 13 09:09:24 2015
;; MSG SIZE  rcvd: 42

Macadamia:~ dwood$ dig www.w3id.org

; <<>> DiG 9.8.3-P1 <<>> www.w3id.org
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 53734
;; flags: qr rd ra; QUERY: 1, ANSWER: 2, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;www.w3id.org.			IN	A

;; ANSWER SECTION:
www.w3id.org.		10799	IN	CNAME	webredir.vip.gandi.net.
webredir.vip.gandi.net.	11090	IN	A	217.70.184.38

;; Query time: 126 msec
;; SERVER: 10.5.50.1#53(10.5.50.1)
;; WHEN: Wed May 13 09:09:30 2015
;; MSG SIZE  rcvd: 82

Macadamia:~ dwood$ 
]]

**Comment by msporny** [comment]
Looking into it...

**Comment by msporny** [comment]
Ok, fixed... maybe - it takes a while for this stuff to propagate, so this time should work. I mistakenly set the IP to DB's public cloud instead of the Rackspace VM IP. Thankfully, @davidlehn fixed it shortly thereafter.

**Comment by msporny** [comment]
Post-mortem on this tragic thread is here: https://lists.w3.org/Archives/Public/public-perma-id/2015May/0003.html

**Comment by davidboweninrupt** [comment]
Apologies for necroposting but as you already know this is coming up for renewal in a few months:

> Registry Expiry Date: 2024-02-14T21:17:29Z

I'm quite sure you'll get notified and get it sorted.

This update was only to mention that Inrupt's systems are monitoring the TLS certificate and domain expiry for w3id.org. Our alarms will go off if it gets to less than 10 days.

(it looks as though [Kingsley's buddylist](https://github.com/perma-id/w3id.org/issues/81#issuecomment-100027917) didn't take off)
